src/compiler/sourcemap.ts

```diff
@@ -263,8 +264,18 @@ namespace ts {
             exit();
         }
 
+        function serializeMappings(): void {
+            const segments: string[] = [];
+            for (let i = 0, len = mappingCharCodes.length; i < len; i += 1024) {
+                segments.push(String.fromCharCode.apply(undefined, mappingCharCodes.slice(i, i + 1024)));

```

Comment:
Out of curiosity, did you try just appending to `mappings` here?  I would have guessed the big win was coming from the batching, rather than the array trick.

Reply:
Well, if just appending to mappings, then I should write a chunk immediately whenever it is larger than some threshold size.

Reply:
I think you mean you want to append `mappingCharCodes` to `mappings` (and then clear it) whenever it hits (e.g.) length 1024.  If that's what you meant, sounds like a neat idea.  If not, can you please elaborate?

Reply:
Yep, that's precisely it, though since the array can scale arbitrarily, I'll let the size be a bit flexible. Pushed the change.